File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpBlocksFindRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpBlocksFindRequest(
        String id,
        List<String> ids,
        Integer chunkRadius,
        Integer sectionRadius,
        String scanMode,
        Integer limit,
        Boolean includeState
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpBlocksFindRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpBlocksFindRequest(
    val id: String,
    val ids: List<String>,
    val chunkRadius: Int,
    val sectionRadius: Int,
    val scanMode: String,
    val limit: Int,
    val includeState: Boolean?
)
